container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java

```diff
@@ -0,0 +1,75 @@
+package com.yahoo.container.handler;
+
+import com.google.inject.Inject;
+import com.yahoo.component.AbstractComponent;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A component which tracks the up/down status of any clusters which should influence
+ * the up down status of this container itself, as well as the separate fact that such clusters are present.
+ *
+ * This is a separate component which has <b>no dependencies</b> such that the status tracked in this
+ * will survive reconfiguration events and inform other components even immediately after a reconfiguration
+ * (where the true statue of clusters may not yet be available).
+ *
+ * @author bratseth
+ */
+public class ClustersStatus extends AbstractComponent {
```

Comment:
Hmm, we haven't had this rule before. AbstractComponent has a lot of stuff about id's which seems unrelated to deconstruction and I don't know what you lose by not implementing it (it is not in the API, but certainly you lose something, otherwise why is it there?), and even the simplest possible id logic is worth implementing just once.

So, if you think we should have this rule I suggest creating a class SimpleAbstractComonent that AbstractComponent adds the deconstruction logic to and that we can also extend directly. It should be done separate from this PR though.

Reply:
We have had that rule since container-di. The ID logic is just remains from pre-di, and should have been removed. A component is just a Pojo that is declared in services or in the model. 

Which reminds me that I think you forgot to add it to the config model (and I didn't notice until now). Please add it, e.g. in the ContainerCluster ctor. Since the class ends up in the `container-disc` bundle, `addSimpleComponent(String)` will do just fine.

Reply:
In what way have we had that rule? For example, I don't see it mentioned in the Javadoc of AbstractComponent, and we have plenty of components which breaks this rule.

Regarding the id logic: Yes, that looks right, the call stacks ends in tests. Let's clean it up before Vespa 7, but separate from this PR plz.

Reply:
Adding the component: Done - thanks! (I always forget about that because I don't want it to exist  :-) )